src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,25 +1,43 @@
 package com.es.phoneshop.model.product;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
+    List<Product> products;
+
+    public ArrayListProductDao() {
+        products = new ArrayList<>();
+    }
+
+
     @Override
     public Product getProduct(Long id) {
-        throw new RuntimeException("Not implemented");
+        return products.stream().
+                filter(product -> product.getId().equals(id)).
+                findAny().orElseThrow( () -> new RuntimeException( "Product is not found"));
     }
 
     @Override
     public List<Product> findProducts() {
-        throw new RuntimeException("Not implemented");
+        return products.stream().
+                filter(product -> product.getPrice() != null && product.getStock() > 0).
+                collect(Collectors.toList());
     }
 
     @Override
     public void save(Product product) {
-        throw new RuntimeException("Not implemented");
+        if (product == null) {
```

Comment:
What happens if there is such product in the list already? Should we check this and throw an exception in this case?

Consider rewriting using streams.